cli/azd/pkg/ai/python_bridge.go

```diff
@@ -0,0 +1,143 @@
+package ai
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/python"
+	"github.com/azure/azure-dev/cli/azd/resources"
+)
+
+// ScriptPath is a type to represent the path of a Python script
+type ScriptPath string
+
+const (
+	// PromptFlowClient is the path to the PromptFlow Client Python script
+	PromptFlowClient ScriptPath = "pf_client.py"
+	// MLClient is the path to the ML Client Python script
+	MLClient ScriptPath = "ml_client.py"
+)
+
+// PythonBridge is an interface to execute python components from the embedded AI resources project
+type PythonBridge interface {
+	Initialize(ctx context.Context) error
+	RequiredExternalTools(ctx context.Context) []tools.ExternalTool
+	Run(ctx context.Context, scriptName ScriptPath, args ...string) (*exec.RunResult, error)
+}
+
+// pythonBridge is a bridge to execute python components from the embedded AI resources project
+type pythonBridge struct {
+	azdCtx        *azdcontext.AzdContext
+	env           *environment.Environment
+	pythonCli     *python.PythonCli
+	commandRunner exec.CommandRunner
+	workingDir    string
+	initialized   bool
+}
+
+// NewPythonBridge creates a new PythonBridge instance
+func NewPythonBridge(
+	azdCtx *azdcontext.AzdContext,
+	env *environment.Environment,
+	pythonCli *python.PythonCli,
+	commandRunner exec.CommandRunner,
+) PythonBridge {
+	return &pythonBridge{
+		azdCtx:        azdCtx,
+		env:           env,
+		pythonCli:     pythonCli,
+		commandRunner: commandRunner,
+	}
+}
+
+// Initialize initializes the PythonBridge
+// Copies embedded AI script files, creates a python virtual environment and installs the required dependencies
+func (b *pythonBridge) Initialize(ctx context.Context) error {
+	if b.initialized {
+		return nil
+	}
```

Comment:
This works only within the same command run, right?
What happens on the second run of the same command?  Do we re-create the python v-environment and install deps again?

Reply:
Yes, but it is fast because the requirements have already been met and they don't get reinstalled.